tls/extensions/s2n_client_key_share.c

```diff
@@ -226,21 +177,19 @@ static int s2n_client_key_share_send(struct s2n_connection *conn, struct s2n_stu
 {
     struct s2n_stuffer_reservation shares_size = {0};
     POSIX_GUARD(s2n_stuffer_reserve_uint16(out, &shares_size));
-
-    if (s2n_is_hello_retry_handshake(conn)) {
-        POSIX_GUARD(s2n_send_hrr_keyshare(conn, out));
-    } else {
-        POSIX_GUARD(s2n_generate_default_pq_hybrid_key_share(conn, out));
-        POSIX_GUARD(s2n_generate_default_ecc_key_share(conn, out));
-    }
-
+    POSIX_GUARD(s2n_generate_default_pq_hybrid_key_share(conn, out));
+    POSIX_GUARD(s2n_generate_default_ecc_key_share(conn, out));
     POSIX_GUARD(s2n_stuffer_write_vector_size(&shares_size));
 
+    /* We must have written at least one share */
+    POSIX_ENSURE(s2n_stuffer_data_available(out) > shares_size.length, S2N_ERR_BAD_KEY_SHARE);
```

Comment:
If it was equal, then we could have just written the size of the shares (0 as a uint16) and nothing else. We need to make sure we've actually written some shares.

Reply:
Can this line be made exact, and be made into this check?
`POSIX_ENSURE(s2n_stuffer_data_available(out) == (sizeof(uint16_t) + shares_size.length), S2N_ERR_BAD_KEY_SHARE);`

Reply:
`shares_size.length` IS the sizeof(uint16_t). It's the size of the space reserved for the size: https://github.com/aws/s2n-tls/blob/main/stuffer/s2n_stuffer.h#L123 I was trying to avoid specifying the size of the size in two different places.

Maybe I should give up on using it that way and just use `POSIX_ENSURE(s2n_stuffer_data_available(out) > sizeof(uint16_t), S2N_ERR_BAD_KEY_SHARE)`?